filters/united-states/massachusetts/cambridge/public-art-json-to-json.js

```diff
@@ -1,35 +1,94 @@
+  if (!std_lib) {
+      throw "standard library not provided";
   }
+  let add_required = std_lib.get("add_required");
+  let add_if_not_null = std_lib.get("add_if_not_null")
+  let remove_if_null = std_lib.get("remove_if_null");
+  let remove_if_empty = std_lib.get("remove_if_empty");
+  let validate_params = std_lib.get("validate_params");
+  let create_dates_template = std_lib.get("create_dates_template");
+  let remove_null_date_fields = std_lib.get("remove_null_date_fields");
 
+  // validate parameters object
+  schema = validate_params(schema, validator);
+
+  // convert JSON data to object form
+  if (typeof data === 'string' || data instanceof String) {
+      data = JSON.parse(data);
+  }
+
+  // define new data and errors array
   let new_data = [];
+  let errors = [];
 
-  data.map((d) => {
-    let item = {};
-    item.name = d.list_title;
-    if (item.name === undefined) {
-      console.log(`Data name not found for art with url ${d.url}`);
-    }
-    let coordinates = {
-      longitude: parseFloat(d.longitude),
-      latitude: parseFloat(d.latitude),
-    };
-    if (
-      coordinates.longitude === undefined ||
-      coordinates.latitude === undefined
-    ) {
-      console.log(`Data coordinates not found for art with url ${d.url}`);
-    }
-    item.coordinates = coordinates;
-    new_data.push(item);
-  });
+  // iterate through each data entry
+  data.map(d => {
+      let item = {};
+      let skip = false;
 
-  if (stringify) {
-    return JSON.stringify(new_data, null);
-  }
+      // add name (required)
+      if (!add_required(item, "name", d, d.list_title, errors)) {
+          skip = true;
+      }
 
-  return new_data;
-};
+      // add coordinates (required)
+      let coordinates = {};
+      if (!add_required(coordinates, "longitude", d, parseFloat(d.longitude), errors)) {
+          skip = true;
+      }
+      if (!add_required(coordinates, "latitude", d, parseFloat(d.latitude), errors)) {
+          skip = true;
+      }
+      item.coordinates = coordinates;
+  
+      add_if_not_null(item, "artist", d.artist_ful);
+      add_if_not_null(item, "material", d.detailed_m);
+      add_if_not_null(item, "type", d.object_typ);
+      add_if_not_null(item, "area", d.neighborho);
+      add_if_not_null(item, "image_url", d.photo_webs);
+      
+      item.dates = create_dates_template();
+      regular_date = d.created_da.split("T")[0]
+      add_if_not_null(item.dates.created, "year", parseInt(regular_date.split("-")[0]));
+      add_if_not_null(item.dates.created, "month", parseInt(regular_date.split("-")[1]));
+      add_if_not_null(item.dates.created, "date", parseInt(regular_date.split("-")[2]));
+      remove_null_date_fields(item)
+      
+      item.image_urls = [];
+      image_urls = [d.photo_webs];
+      for (let i = 0; i < image_urls.length; i++) {
+        if (image_urls[i] != null) {
+          item.image_urls.push(image_urls[i])
+        }
+      }
```

Comment:
it may be good to check if image.image_urls is empty (ie all image urls are null), then you should delete this field from item